src/utils/mixins.js

```diff
@@ -32,4 +32,48 @@ const accessLevelChecker = {
   },
 };
 
-export { accessLevelChecker, sizeFormatter };
+/**
+ * Provides read/write-style lock with non-reactive components such that
+ * the lock can be used as a side-effect in (especially async) computed properties.
+ * Includes debouncing.
```

Comment:
It took me a while to understand exactly why this was being used. I think part of that is due to the name, "lock" is a little misleading IMO, this is really a non-reactive counter (intentionally designed to circumvent reactivity on the counter itself) that debounces the setting of a reactive flag. Even in this use case, it's not really "locking" anything.

I can't decide whether or not I think this is too early to generalize into a mixin, do you have a strong feeling one way or the other? (Related, I think `formatUserName` might actually be a good candidate for a mixin...)

Reply:
The mixin is so-named because it's basically the read- part of a reader-writer-lock.  The debounce part does detract from that though, so it's not usable as an actual resource lock.

The behavior of the mixin is generic enough to separate.  Its only conceivable purpose is handling render of busy spinners and other graphics.  I don't feel that strongly about it though.   If we find it can be recycled, we can refactor it then.

I would propose:
*  renaming it to debounceCounter, with methods `inc()` and `dec()`, and a Boolean data property `isZero`.

Otherwise, pulling it into search with the same method names.

Reply:
I do like the name `debounceCounter` better. Just throwing stuff out there without fully thinking it through, what if we did it as a self-contained class rather than a mixin?

```javascript
class DebounceCounter extends Vue {
    constructor(delay=200) {
        super({
          data: {
            flag: false,
          },
        });
        this.delay = 200;
        this.count = 0;
    }

    inc() ...
    dec() ...
    _toggleFlag(val) ...
```

That encapsulates things nicely behind an abstraction rather than leaking several names into the components that use it. @jbeezley @matthewma7 @subdavis any thoughts? Better suggestions?

Reply:
(It also has the benefit of allowing a component to potentially use multiple debouncing counters that are independent, which the mixin would not)